lesson21/main.js

```diff
@@ -118,24 +121,100 @@ const formingTableHeaderNameWithKey = (key) => {
 
 const createTableData = (usersData) => {
   const fragment = document.createDocumentFragment();
-  usersData.forEach((userData) => {
-    const trOfTdata = document.createElement("tr");
+  usersData.forEach((userData, index) => {
+    const trOfTdata = createElementWithClassName("tr", "users-table__tr-td");
+    const usersIndex = ++index;
     /**
      *Extract value from each of usersData
      * @param {Object} userData the object of each users data
      * @param {String,Number} val value of each users data
      */
 
-    Object.values(userData).forEach((val) => {
+    Object.keys(userData).forEach((key) => {
       const td = createElementWithClassName("td", "users-table__td");
-      td.textContent = val;
+      key === "id"
+        ? (td.textContent = usersIndex)
+        : (td.textContent = userData[key]);
       trOfTdata.appendChild(td);
     });
     fragment.appendChild(trOfTdata);
   });
   return fragment;
 };
 
+const createSortBtn = () => {
+  const btn = createElementWithClassName("button", "sort-btn");
+  const sortArrow = createElementWithClassName("img", "sort-img");
+  // bothのイメージをデフォルトとする
+  sortArrow.src = "./img/both.svg";
+  btn.appendChild(sortArrow);
+  return btn;
+};
+
+const renderSortBtn = () => {
+  const btn = createSortBtn();
+  const targetEle = document.getElementById("id");
+  targetEle.appendChild(btn);
+};
+
+const clickSortBtn = () => {
+  const sortArrow = document.querySelector(".sort-btn");
+  sortArrow.addEventListener("click", () => {
+    changeSortStateAndArrowImg(sortArrow);
+    rerenderTableData(newUsersData);
+  });
+};
+
+const getNewUsersData = () => {
+  const trOfTdata = document.querySelectorAll(".users-table__tr-td");
+  return (newUsersData = Array.from(trOfTdata));
+};
+
+const changeSortStateAndArrowImg = (sortArrow) => {
+  if (sortState === "both") {
+    sortState = "asc";
+    sortArrow.src = "/img/asc.svg";
+  } else if (sortState === "asc") {
+    sortState = "desc";
+    sortArrow.src = "/img/desc.svg";
+  } else {
+    sortState = "both";
+    sortArrow.src = "/img/both.svg";
+  }
+};
```

Comment:
MUST
The function here does not seem to be working properly. I pressed the button several times, but it stayed both.
<img width="665" alt="スクリーンショット 2022-01-06 10 22 31" src="https://user-images.githubusercontent.com/43788628/148312956-0c83901a-0a8a-46ef-9260-7791a5b0ccb9.png">

Reply:
It's because of this commit [0fc5ec2](https://github.com/yuka830/js-lessons/pull/27/commits/0fc5ec2b3ce225f50ed9a1243ac5dbc32119fcbb) and I didn't realized...sorry.
I've fixed. Thank you!
[9b19410](https://github.com/yuka830/js-lessons/pull/27/commits/9b19410c583eb904d6dee2cf6c4562a537573d78)